vpt/sources.py

```diff
@@ -0,0 +1,45 @@
+'''Manages obtaining the available input/output sources.'''
+from typing import List
+from textwrap import shorten
+
+import sounddevice as sd
+
+def get_default_audio_input_name() -> str:
+    '''Gets the name of the default audio input device.'''
+    return sd.query_devices(kind='input')['name']
+
+def list_audio_inputs() -> List[dict]:
+    '''Returns a list of all audio input devices.'''
+    devices = sd.query_devices()
+    default = sd.query_devices(kind='input')
+    devices = [device for device in devices if device['max_input_channels'] > 0]
+    for device in devices:
+        if device == default:
+            device['default'] = True
+        else:
+            device['default'] = False
+    return devices
+
+def print_audio_inputs():
+    '''Prints the available input devices in a nice table.'''
+    devices = list_audio_inputs()
+    print(f'┌{"-" * 52}┬{"-" * 12}┬{"-" * 14}┐')
+    print(f'| {"Name":^50} | {"Channels":^10} | {"Sample rate":^12} |')
+    print(f'├{"-" * 52}┼{"-" * 12}┼{"-" * 14}┤')
+    for device in devices:
+        if device['default']:
+            device['name'] = '* ' + device['name']
+        device['name'] = shorten(device['name'], width=50, placeholder='..')
+        print(f'| {device["name"]:<50} | {device["max_input_channels"]:^10} '
+              f'| {device["default_samplerate"]:^12} |')
+    print(f'└{"-" * 52}┴{"-" * 12}┴{"-" * 14}┘')
+    print('* = default')
+
+
+def list_video_inputs():
+    '''Returns a list of all video input devices (a.k.a. cameras)'''
```

Comment:
For Linux: https://askubuntu.com/a/848390/601580
For Windows: https://github.com/yushulx/python-capture-device-list

Even if it is behind dependencies and complicated build processes for now, I think it'd be better to have it working for the sprint

Reply:
Oh, didn't notice it was a draft PR. Well, consider this some aid in research :p

Reply:
And how would you execute that command on Linux? `os.system`? If you could implement the Linux variant, I would appreciate it.
And for Windows, this complicates the installation so much, with all the installation of Windows SDK or Visual Studio (I don't even understand what has to be installed), and then using that path (which depends on each user, not just the OS) inside the Python code.

Reply:
Seems like there are Python bindings for this `v4l2` thing: https://pypi.org/project/v4l2/

Reply:
Problem is, I don't really have a webcamera, so even `ls /dev/video*` fails because there isn't a single input device :)
@IamMaxim maybe you could help out with this?